extension/agenthealth/handler/stats/agent/agent.go

```diff
@@ -80,6 +82,45 @@ func (s *Stats) Merge(other Stats) {
 	if other.EntityRejected != nil {
 		s.EntityRejected = other.EntityRejected
 	}
+	if other.StatusCodes != nil {
```

Comment:
I can do something like this below:

```
	if other.StatusCodes == nil {
		return
	}

	if s.StatusCodes == nil {
		s.StatusCodes = make(map[string][5]int)
	}

	for key, value := range other.StatusCodes {
		if existing, ok := s.StatusCodes[key]; ok {
			s.StatusCodes[key] = [5]int{
				existing[0] + value[0], // 200
				existing[1] + value[1], // 400
				existing[2] + value[2], // 408
				existing[3] + value[3], // 413
				existing[4] + value[4], // 429
			}
		} else {
			s.StatusCodes[key] = value
		}
	}
```

but it will be different from all the other if statements so I think I'm going to leave as is.